### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by drewnoakes** [tom-englert/ProjectConfigurationManager#18]
This looks amazing, however it's incredibly slow on my solution, to the points where I do not think I can really use it. I have 88 projects, of various kinds. There are some [reviews](https://marketplace.visualstudio.com/items?itemName=TomEnglert.ProjectConfigurationManager#review-details) that suggest others are having performance problems as well.

If you're unable to reproduce this, let me know how I can help.

**Comment by tom-englert** [comment]
@drewnoakes yes, performance is not stunning, but this is mostly due to the fact that it shows so many information at once and it has to get all the information it shows out of VS. 
With one of my larger solutions (86 projects) it may take a few seconds to switch a tab, but for me that's acceptable.

If you want to help to improve this, you could do some performance profiling and help to identify some bottle necks.

**Comment by drewnoakes** [comment]
I was seeing around 30 seconds to load the UI initially (during which time the entire UI thread is blocked). Changing tabs takes around 20 seconds. Scrolling the UI also takes several seconds.

Am back at work today but could find some time to provide profiling results. Do you have any tips on profiling VS add ins?

**Comment by tom-englert** [comment]
Just run another instance of VS, then attach the profiler to the running process. (devenv.exe)

**Comment by tom-englert** [comment]
@drewnoakes I always keep our solution configurations to only include one debug and one release configuration. 
Unfortunately VS tends to litter the solution with various combinations of "Mixed", "x86", etc.. which are usually unneeded and confusing, but can significantly slow down everything. 
I worked on such a project recently, and after cleaning up the unused configurations, the response times have improved significantly.

**Comment by dfoulk** [comment]
Yeah, this extension is not really usable with Xamarin.Forms solutions in VS2017. Running on an i7 box with plenty of memory and it's locking up my UI thread for 9+ seconds whenever I do anything. It looks like it would otherwise be a useful extension man. I know how difficult it can be to make VS extensions. I've made a couple for our team here at work.

Is the issue caused by the extension accessing the .SLN file while it's open? There's not that much data here. Not enough to choke a modern PC, right?

**Comment by dfoulk** [comment]
Thank you for taking the time to share this with us though @tom-englert. It's much appreciated!
